packages/api-server/api_server/repositories/tasks.py

```diff
@@ -242,6 +243,34 @@ async def get_task_state(self, task_id: str) -> Optional[TaskState]:
             return None
         return TaskState(**result.data)
 
+    async def query_task_queue_entry(
+        self, query: QuerySet[DbTaskState], pagination: Optional[Pagination] = None
+    ) -> List[TaskQueueEntryPydantic]:
+        try:
+            if pagination:
+                query = add_pagination(query, pagination)
+            # TODO: enforce with authz
+            results = await query.values(
+                "id_",
+                "assigned_to",
+                "unix_millis_start_time",
+                "unix_millis_finish_time",
+                "status",
+                "unix_millis_request_time",
+                "requester",
+                "pickup",
+                "destination",
+            )
+            entries = []
+            for r in results:
+                status = r["status"]
+                if "Status." in status:
+                    r["status"] = r["status"].split("Status.")[1]
+                entries.append(TaskQueueEntryPydantic(**r))
```

Comment:
I think we should avoid massaging the data for perf and consistency reasons, you can massage it on the frontend instead. Or at least refactor it so it is consistent, e.g., add the option to filter values to `query_task_states` and just have `query_task_queue_entry` call it with the filters.

Reply:
Gotcha, I've moved the massaging to the frontend now